github/File.py

```diff
@@ -29,110 +29,76 @@
 ################################################################################
-
 from typing import Any, Dict
 
-import github.GithubObject
+from github.GithubObject import Attribute, NonCompletableGithubObject, NotSet
 
 
-class File(github.GithubObject.NonCompletableGithubObject):
+class File(NonCompletableGithubObject):
     """
     This class represents Files
     """
 
-    def __repr__(self):
+    def _initAttributes(self) -> None:
+        self._additions: Attribute[int] = NotSet
+        self._blob_url: Attribute[str] = NotSet
+        self._changes: Attribute[int] = NotSet
+        self._contents_url: Attribute[str] = NotSet
+        self._deletions: Attribute[int] = NotSet
+        self._filename: Attribute[str] = NotSet
+        self._patch: Attribute[str] = NotSet
+        self._previous_filename: Attribute[str] = NotSet
+        self._raw_url: Attribute[str] = NotSet
+        self._sha: Attribute[str] = NotSet
+        self._status: Attribute[str] = NotSet
+
+    def __repr__(self) -> str:
         return self.get__repr__({"sha": self._sha.value, "filename": self._filename.value})
 
     @property
-    def additions(self):
-        """
-        :type: integer
-        """
+    def additions(self) -> int:
         return self._additions.value
 
     @property
-    def blob_url(self):
-        """
-        :type: string
-        """
+    def blob_url(self) -> str:
         return self._blob_url.value
 
     @property
-    def changes(self):
-        """
-        :type: integer
-        """
+    def changes(self) -> int:
         return self._changes.value
 
     @property
-    def contents_url(self):
-        """
-        :type: string
-        """
+    def contents_url(self) -> str:
         return self._contents_url.value
 
     @property
-    def deletions(self):
-        """
-        :type: integer
-        """
+    def deletions(self) -> int:
         return self._deletions.value
 
     @property
-    def filename(self):
-        """
-        :type: string
-        """
+    def filename(self) -> str:
         return self._filename.value
 
     @property
-    def patch(self):
-        """
-        :type: string
-        """
+    def patch(self) -> str:
         return self._patch.value
 
     @property
-    def previous_filename(self):
-        """
-        :type: string
-        """
+    def previous_filename(self) -> str:
         return self._previous_filename.value
 
     @property
-    def raw_url(self):
-        """
-        :type: string
-        """
+    def raw_url(self) -> str:
         return self._raw_url.value
 
     @property
-    def sha(self):
-        """
-        :type: string
-        """
+    def sha(self) -> str:
         return self._sha.value
 
     @property
-    def status(self):
-        """
-        :type: string
-        """
+    def status(self) -> str:
         return self._status.value
 
-    def _initAttributes(self) -> None:
```

Comment:
Why is this moved? let's keep the dif minimal for easier review...

Reply:
EnricoMi think this is better

Reply:
@borda you are right, minimal diffs are better in general. In this particular case, having all attributes and types defined at the top of the class makes it more comprehensible, so while we are touching all files anyway, we are moving this method up.